pkg/lib/queueinformer/queueinformer_operator.go

```diff
@@ -194,6 +199,16 @@ func (o *operator) start(ctx context.Context) error {
 	go func() {
 		defer close(errs)
 		v, err := o.serverVersion.ServerVersion()
+		if err == nil {
+			o.logger.Infof("connection established. cluster-version: %v", v)
+			return
+		}
+		select {
+		case <-time.After(defaultServerVersionInterval):
+		case <-ctx.Done():
+			return
+		}
+		v, err = o.serverVersion.ServerVersion()
```

Comment:
So if I'm reading this correctly, we attempt to get the server version twice. If the first attempt fails, we retry after the `defaultServerVersionInterval`. If we ever succeed we return the serverVersion. Could you share your reasoning why we make two attempts here but infinite attempts above?

Reply:
Unit tests... the unit tests actually wait for this initialization to complete (success or failure), which is guaranteed to fail.

Reply:
(I would've liked to have had an infinite wait, but then the unit-tests timeout)